ably/ably_client.go

```diff
@@ -1,12 +1,25 @@
 package main
 
 import (
+	"time"
+
 	"github.com/ably/ably-go/ably"
 )
 
-func newAblyClient(testConfig TestConfig) (*ably.RealtimeClient, error) {
+// newAblyClient creates a new Ably realtime client, making multiple attempts
+// to avoid transient connection errors being reported as failures.
+func newAblyClient(testConfig TestConfig) (client *ably.RealtimeClient, err error) {
 	options := ably.NewClientOptions(testConfig.APIKey)
 	options.Environment = testConfig.Env
 
-	return ably.NewRealtimeClient(options)
+	timeout := 30 * time.Second
+	delay := 100 * time.Millisecond
```

Comment:
I think this could be 1s. I'm be worried about the DoS effect of this if the client is retrying aggressively with no backoff.

Reply:
I've added a backoff in df47e62 which starts with 100ms (i.e. pretty much straight away for the second attempt), but then 1s, 5s, 10s, 10s before giving up. WDYT?